### DISCUSSION TITLE
Hole in floor

### DISCUSSION THREAD

**Discussion by ocmino** [pmndrs/xr#245] [Q&A] [answered]
Hello everyone!

I want to display a hole in the ground with AR, but have no idea on how to do that. I have a simple pool shaped mesh that I can display in ar, but that's  as far as I have come. Any help? Thanks!

**Comment by saitonakamura** [comment]
Do you have a concept of how it would look in a simple 3d scene?

**Reply by ThomasRutzer** [reply]
I made this [experiment](https://github.com/ThomasRutzer/webxr-image-targets-hole) once…It's not up to date, but maybe gives you some pointers…

**Comment by aouaiti** [comment]
I would use drei helpers "Mask" or "MeshPortalMaterial"

**Comment by bbohlender** [comment, marked as answer]
You want to render the hole using as a cylinder that only renders the inside
```tsx
<mesh>
  <meshPhongMaterial side={BackSide} />
  <cylinderGeometry/>
</mesh>
```

and place this cylinder with the top side at `0,0,0`.
Now, in are the hole is already visible but you can see it from the side, so you want to make sure its only rendered when looking into the whole.

This can be achieved by rendering a plane that has a hole cut into at the position and size of the cylinder. You can build this plane with blender for instance. Alternatively, use a stencil buffer to cut out the whole into the plane.

Once you have this plane, import it as a gltf for instance and retrieve the geometry from the gltf. Using the `loadedPlaneGeometry` we can render the plane but not draw its color. This allows to prevent rendering the cylinder from the side, while still showing the passthrough where the plane is. 

```tsx
<mesh geometry={loadedPlaneGeometry}>
  <meshBasicMatererial colorWrite={false} renderOrder={-100} />
</mesh>
```

I hope it helps :)
